tools/src/main/java/org/apache/kafka/tools/ConnectPluginPath.java

```diff
@@ -0,0 +1,399 @@
+                    Set<ManifestEntry> classpathManifests = findManifests(classpathSource);
+                    classpathManifests.forEach(isolatedManifests::remove);
+                    Map<String, Set<ManifestEntry>> indexedManifests = indexManifests(isolatedManifests);
+                    Map<String, List<String>> aliases = invertAliases(PluginUtils.computeAliases(merged));
+                    Map<String, Set<ManifestEntry>> unloadablePlugins = new HashMap<>(indexedManifests);
+                    merged.forEach(pluginDesc -> unloadablePlugins.remove(pluginDesc.className()));
+                    merged.forEach(pluginDesc -> operation(config, isolatedSource, aliases, indexedManifests, pluginDesc.className(), pluginDesc.version(), pluginDesc.type(), true));
+                    for (Set<ManifestEntry> manifestEntries : unloadablePlugins.values()) {
+                        manifestEntries.forEach(manifestEntry -> operation(config, isolatedSource, aliases, indexedManifests, manifestEntry.className, PluginDesc.UNDEFINED_VERSION, manifestEntry.type, false));
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static <T> void operation(
+            // fixed
+            Config config,
+            // each location
+            PluginSource source,
+            Map<String, List<String>> aliases,
+            Map<String, Set<ManifestEntry>> manifests,
+            // each plugin
+            String pluginName,
+            String pluginVersion,
+            PluginType pluginType,
+            boolean loadable
+    ) {
+        operation(config, source, pluginName, pluginVersion, aliases.getOrDefault(pluginName, Collections.emptyList()), pluginType, loadable, manifests.containsKey(pluginName));
+    }
+
+    private static <T> void operation(
+            // fixed
+            Config config,
+            // each location
+            PluginSource source,
+            // each plugin
+            String pluginName,
+            String pluginVersion,
+            List<String> aliases,
+            PluginType pluginType,
+            boolean loadable,
+            boolean hasManifest
+    ) {
+        Path pluginLocation = source.location();
+        if (config.command == Command.LIST) {
+            String firstAlias = aliases.size() > 0 ? aliases.get(0) : "null";
+            String secondAlias = aliases.size() > 1 ? aliases.get(1) : "null";
+            // there should only be one location, otherwise the plugin source column will be ambiguous.
+            config.out.printf(
+                    "%s\t%s\t%s\t%s\t%s\t%s\t%b\t%b%n",
+                    pluginLocation,
+                    pluginName,
+                    firstAlias,
+                    secondAlias,
+                    pluginVersion,
+                    pluginType,
+                    loadable,
+                    hasManifest
+            );
```

Comment:
The format string here is a little tricky to read, and a little brittle. We can use some Java 8 streams logic to make things a bit more flexible:
```suggestion
            String pluginInfo = Stream.of(
                    pluginLocation,
                    pluginName,
                    firstAlias,
                    secondAlias,
                    pluginVersion,
                    pluginType,
                    loadable,
                    hasManifest
            ).map(Objects::toString).collect(Collectors.joining("\t"));
            // there should only be one location, otherwise the plugin source column will be ambiguous.
            config.out.println(pluginInfo);
```